### PR TITLE
Create GroupMembership block condition plugin

### PR DESCRIPTION
**Pull request by andyg5000**
Create the ability to show/hide Drupal blocks based on the users membership in the group loaded from OG context.

### CODE REVIEW

File: src/Plugin/Condition/GroupMembership.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    $user = $this->getContextValue('user');
+    $membership = Og::getMembership($group, $user);
+
+    // There is no membership.
+    if (empty($membership)) {
+      return FALSE;
+    }
+
+    // The membership status is not active.
+    if ($membership->getState() != OgMembership::STATE_ACTIVE) {
```
Comment (MPParsley) [inline src/Plugin/Condition/GroupMembership.php]:
```suggestion
    if ($membership->getState() !== OgMembership::STATE_ACTIVE) {
```

File: src/Plugin/Condition/GroupMembership.php [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+ * @Condition(
+ *   id = "og_group_membership",
+ *   label = @Translation("Group Membership"),
+ *   context = {
+ *     "og" = @ContextDefinition("entity", label = @Translation("Group")),
+ *     "user" = @ContextDefinition("entity:user", label = @Translation("User")),
+ *   }
+ * )
+ */
+class GroupMembership extends ConditionPluginBase {
```
Comment (MPParsley) [inline src/Plugin/Condition/GroupMembership.php:23]: Perhaps we can rename this to OgGroupMembership.

**Review by MPParsley — APPROVED** [review]

**Comment by MPParsley** [comment]
Should we group the group conditions in one plugin/tab? See #265

block-visibility-plugin

File: src/Plugin/Condition/GroupMembership.php [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+use Drupal\og\Entity\OgRole;
+use Drupal\og\Og;
+
+/**
+ * Provides an 'OG Membership' block visibility condition.
+ *
+ * @Condition(
+ *   id = "og_group_membership",
+ *   label = @Translation("Group Membership"),
+ *   context = {
```
Comment (MPParsley) [inline src/Plugin/Condition/GroupMembership.php]: Providing context definitions via the "context" key is deprecated in Drupal 8.7.x and will be removed before Drupal 9.0.0. Use the "context_definitions" key instead.

```suggestion
 *   context_definitions = {
```

File: tests/src/Kernel/GroupMembershipConditionTest.php [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    $this->conditionManager = $this->container->get('plugin.manager.condition');
+    $this->groupTypeManager = $this->container->get('og.group_type_manager');
+
+    // Add membership and config schema.
+    $this->installConfig(['og']);
+    $this->installEntitySchema('entity_test');
+    $this->installEntitySchema('node');
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('og_membership');
+    $this->installSchema('system', ['queue', 'sequences']);
```
Comment (MPParsley) [inline tests/src/Kernel/GroupMembershipConditionTest.php]: Special handling of system module schemas in \Drupal\KernelTests\KernelTestBase::installSchema has been deprecated in Drupal 8.7.x, remove any calls to this method that use invalid schema names. See https://www.drupal.org/node/3003360.

```suggestion
    $this->installSchema('system', ['sequences']);
```

File: src/Plugin/Condition/GroupMembership.php [hunk 5]
```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Drupal\og\Plugin\Condition;
```
Comment (MPParsley) [inline src/Plugin/Condition/GroupMembership.php:6]:
```suggestion
declare(strict_types = 1);

declare(strict_types = 1);
namespace Drupal\og\Plugin\Condition;
```

File: tests/src/Kernel/GroupMembershipConditionTest.php [hunk 6]
```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace Drupal\Tests\og\Kernel;
```
Comment (MPParsley) [inline tests/src/Kernel/GroupMembershipConditionTest.php:5]:
```suggestion
declare(strict_types = 1);

namespace Drupal\Tests\og\Kernel;
```

**Comment by claudiu-cristea** [comment]
Since https://www.drupal.org/project/og/issues/3467694 was merged into the `8.x-1.x` branch, the development has been moved to https://www.drupal.org/project/og. If this change is still actual, please open an issue at https://www.drupal.org/project/issues/og and provide there a _merge request_ similar with this pull request.

Thank you for contributing to Organic groups module.

**Pull request closed without merging**